tests/apollo/test_skvbc_preexecution.py

```diff
@@ -256,8 +255,9 @@ async def test_pre_execution_with_added_constant_load(self, bft_network, skvbc,
         This test validates that pre-execution and normal execution coexist correctly.
         """
         bft_network.start_all_replicas()
-        num_preexecution_requests = 200
+        await trio.sleep(SKVBC_INIT_GRACE_TIME)
```

Comment:
Is there a way to poll this instead? Maybe an an atomic variable or something, or a sample request can be sent.

I just really hate sleeps, and this looks like one that could cause flaky test runs.

Reply:
Well, not sure what exactly to poll for... 

This problem is specific to how the pre-processor is initialized.

I generally agree with your comment, but here the system is simply not up yet, so I think it makes sense to wait.

Another option would be to try running a pre-execution request until it succeeds, but it would create an undesired entry in the blockchain...

Reply:
You're waiting for the thread pool to start up right? Can you use some sort of out of band mechanism? If everything was in C++ you could poll on an atomic or use a condition variable. But for testing, maybe you just instead publish an environment variable or write a file to be read. There has to be a better way than just using an arbitrary sleep.

Now the question is whether what I mentioned is actually worth the effort. In most cases the answer is likely yes, but in this case maybe not. So if not, how long to do you make the timeout? And what happens if it's not long enough?

Reply:
A timeout of `SKVBC_INIT_GRACE_TIME=2` (seconds) seems to be sufficient (used in other pre-execution tests too).